e2e/testcases/helm_sync_test.go

```diff
@@ -187,6 +187,90 @@ func TestHelmDefaultNamespace(t *testing.T) {
 	}
 }
 
+// TestHelmLatestVersion verifies the Config Sync behavior for helm charts when helm.spec.version is not specified. The helm-sync
+// binary should pull down the latest available version in this case. It also tests that if a new helm chart gets pushed, the
+// chart version gets automatically updated by Config Sync.
+// This test will work only with following pre-requisites:
+// Google service account `e2e-test-ar-reader@${GCP_PROJECT}.iam.gserviceaccount.com` is created with `roles/artifactregistry.reader` for accessing images in Artifact Registry.
+// A JSON key file is generated for this service account and stored in Secret Manager
+func TestHelmLatestVersion(t *testing.T) {
+	nt := nomostest.New(t,
+		nomostesting.SyncSource,
+		ntopts.Unstructured,
+		ntopts.RequireGKE(t),
+	)
+
+	rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
+	nt.T.Log("Fetch password from Secret Manager")
+	key, err := gitproviders.FetchCloudSecret("config-sync-ci-ar-key")
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Create secret for authentication")
+	_, err = nt.Shell.Kubectl("create", "secret", "generic", "foo", fmt.Sprintf("--namespace=%s", v1.NSConfigManagementSystem), "--from-literal=username=_json_key", fmt.Sprintf("--from-literal=password=%s", key))
```

Comment:
Could this test use workload identity instead (with `e2e-test-ar-reader`)?

Doesn't seem like we're specifically testing token auth here, and I think it would be preferred to use WI as its easier to manage. The token auth with the exported key is more of an edge case IMO and is harder to manage. I'd suggest we stick to just using the exported token for tests which are specifically verifying token auth.

Reply:
Yup makes sense, updated.

Reply:
it looks like the test is still using token auth

Reply:
Are you possibly looking at the first commit? I changed auth type in the latest one https://github.com/GoogleContainerTools/kpt-config-sync/pull/573/commits/96a6f44b35f2ab008d1775473d66647d21232d11

Reply:
Ah you're right looks like I was looking at a stale commit, looks good now!